paimon-format/src/main/java/org/apache/paimon/format/parquet/filter/ParquetFilters.java

```diff
@@ -0,0 +1,360 @@
+            }
+
+            return FilterApi.and(preds[0].toParquetPredicate(), preds[1].toParquetPredicate());
+        }
+
+        @Override
+        public String toString() {
+            return "AND(" + Arrays.toString(preds) + ")";
+        }
+    }
+
+    /** An Or predicate that can be evaluated by the Parquet Reader. */
+    public static class Or extends ParquetFilters.Predicate {
+        private final ParquetFilters.Predicate[] preds;
+
+        /**
+         * Creates an OR predicate.
+         *
+         * @param predicates The disjunctive predicates.
+         */
+        public Or(ParquetFilters.Predicate... predicates) {
+            this.preds = predicates;
+        }
+
+        @Override
+        public String toString() {
+            return "OR(" + Arrays.toString(preds) + ")";
+        }
+
+        @Override
+        public FilterPredicate toParquetPredicate() {
+            if (preds.length != 2) {
+                throw new RuntimeException("Illegal and children: " + preds.length);
+            }
+
+            return FilterApi.or(preds[0].toParquetPredicate(), preds[1].toParquetPredicate());
+        }
+    }
+
+    private static FilterPredicate predicate(
+            ParquetFilters.Predicate predicate,
+            DataType dataType,
+            String name,
+            Serializable literal) {
+        switch (dataType.getTypeRoot()) {
+            case BOOLEAN:
+                Operators.BooleanColumn col = FilterApi.booleanColumn(name);
+                if (predicate instanceof Equals) {
+                    return FilterApi.eq(col, toParquetObject(dataType, literal));
+                } else if (predicate instanceof Not) {
+                    return FilterApi.eq(col, toParquetObject(dataType, literal));
+                }
+                break;
+            case INTEGER:
+            case TINYINT:
+            case SMALLINT:
+            case TIME_WITHOUT_TIME_ZONE:
+            case DATE:
+                Operators.IntColumn intColumn = FilterApi.intColumn(name);
+                return pred(predicate, intColumn, toParquetObject(dataType, literal));
+            case FLOAT:
+                return pred(
+                        predicate, FilterApi.floatColumn(name), toParquetObject(dataType, literal));
+            case DOUBLE:
+                return pred(
+                        predicate,
+                        FilterApi.doubleColumn(name),
+                        toParquetObject(dataType, literal));
+            case BIGINT:
+                return geLongtFilterPredicate(predicate, dataType, name, literal);
+            case TIMESTAMP_WITHOUT_TIME_ZONE:
+            case TIMESTAMP_WITH_LOCAL_TIME_ZONE:
+                TimestampType timestampType = (TimestampType) dataType;
+                int precision = timestampType.getPrecision();
+                if (precision > 6) {
+                    return getBinaryFilterPredicate(predicate, dataType, name, literal);
```

Comment:
Is this correct? `TIMESTAMP_WITHOUT_TIME_ZONE` compare binary looks not correct.